File: chess-api/src/main/java/com/chess/ChessBoard.java
package com.chess;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class ChessBoard {
    private static final int BOARD_LENGTH = 8;
    private List<Piece> pieces = new ArrayList<Piece>();

    public ChessBoard() {
        pieces.add(new Piece(PieceType.ROOK, 1, 1));
        pieces.add(new Piece(PieceType.KNIGHT, 2, 1));
        pieces.add(new Piece(PieceType.BISHOP, 3, 1));
        pieces.add(new Piece(PieceType.QUEEN, 4, 1));
        pieces.add(new Piece(PieceType.KING, 5, 1));
        pieces.add(new Piece(PieceType.BISHOP, 6, 1));
        pieces.add(new Piece(PieceType.KNIGHT, 7, 1));
        pieces.add(new Piece(PieceType.ROOK, 8, 1));
        for (int x = 1; x <= BOARD_LENGTH; x++) {
            pieces.add(new Piece(PieceType.PAWN, x, 2));
            pieces.add(new Piece(PieceType.PAWN, x, 7));
        }
        pieces.add(new Piece(PieceType.ROOK, 1, 8));
        pieces.add(new Piece(PieceType.KNIGHT, 2, 8));
        pieces.add(new Piece(PieceType.BISHOP, 3, 8));
        pieces.add(new Piece(PieceType.QUEEN, 4, 8));
        pieces.add(new Piece(PieceType.KING, 5, 8));
        pieces.add(new Piece(PieceType.BISHOP, 6, 8));
        pieces.add(new Piece(PieceType.KNIGHT, 7, 8));
        pieces.add(new Piece(PieceType.ROOK, 8, 8));
    }

    public Piece getPiece(int x, int y) {
        for (Piece piece : pieces) {
            if (piece.getX() == x && piece.getY() == y) {
                return piece;
            }
        }
        return null;
    }
}

File: chess-front/src/app/chessBoard.ts
import { Position } from './position';

export class ChessBoard {
  pieces: string[][] = [];
  highlightedSquares: Position[] = [];

  constructor() {
    this.initializeChessBoard();
  }

  getPiece(position: Position): string {
    return this.pieces[position.y][position.x];
  }

  setPiece(pieceName: string, position: Position) {
    this.pieces[position.y][position.x] = pieceName;
  }

  private initializeChessBoard(): void {
    for (let i = 0; i < 8; i++) {
      this.pieces[i] = [];
      for (let j = 0; j < 8; j++) {
        // Initial chess pieces setup can be adjusted based on your needs
        this.pieces[i][j] = this.getInitialChessPiece(i, j);
      }
    }
  }
  private getInitialChessPiece(rowIndex: number, colIndex: number): string {
    //static board with the initial position of pieces
    const row = rowIndex;

    if (row === 0) {
      switch (colIndex) {
        case 0:
        case 7:
          return 'white-rook';
        case 1:
        case 6:
          return 'white-knight';
        case 2:
        case 5:
          return 'white-bishop';
        case 3:
          return 'white-queen';
        case 4:
          return 'white-king';
      }
    } else if (row === 1) {
      return 'white-pawn';
    } else if (row === 6) {
      return 'black-pawn';
    } else if (row === 7) {
      switch (colIndex) {
        case 0:
        case 7:
          return 'black-rook';
        case 1:
        case 6:
          return 'black-knight';
        case 2:
        case 5:
          return 'black-bishop';
        case 3:
          return 'black-queen';
        case 4:
          return 'black-king';
      }
    }
    return ''; // for squares without any pieces
  }
}
